packages/teleport/src/Kubes/AddKube/AddKube.tsx

```diff
@@ -0,0 +1,248 @@
+            }
+            <Validation>
+              {({ validator }) => (
+                <Flex alignItems="center" flexDirection="row">
+                  <form
+                    onSubmit={e => handleGenerate(e, validator)}
+                    style={{ width: '100%' }}
+                  >
+                    <FieldInput
+                      mb={2}
+                      rule={(val: string) => requiredField(val, 'Namespace')}
+                      label="Namespace"
+                      autoFocus
+                      value={namespace}
+                      placeholder="teleport"
+                      width="100%"
+                      mr="3"
+                      onChange={e => setNamespace(e.target.value)}
+                    />
+                    <FieldInput
+                      mb={2}
+                      rule={(val: string) =>
+                        requiredField(val, 'Kubernetes Cluster Name')
+                      }
+                      label="Kubernetes Cluster Name"
+                      labelTip="Name shown to Teleport users connecting to the cluster."
+                      value={clusterName}
+                      placeholder="my-cluster"
+                      width="100%"
+                      mr="3"
+                      onChange={e => setClusterName(e.target.value)}
+                    />
+                    <ButtonPrimary
+                      block
+                      mt="1"
+                      disabled={attempt.status === 'processing'}
+                      type="submit"
+                    >
+                      {cmd ? 'Regenerate Script' : 'Generate Script'}
+                    </ButtonPrimary>
+                  </form>
+                </Flex>
+              )}
+            </Validation>
+          </Box>
+          {cmd && (
+            <Box mb={4}>
+              <Text bold as="span">
+                Step 3
+              </Text>
+              {' - Install the helm chart'}
+
+              <Box>
+                <Text mt="2" mb="1">
+                  The token will be valid for{' '}
+                  <Text bold as={'span'}>
+                    {token.expiryText}.
+                  </Text>
+                </Text>
+                <TextSelectCopy text={cmd} mb={2} />
+                <Text>
+                  <Text as="span" bold>
+                    Tip
+                  </Text>
+                  : Save the YAML file to apply updates later
+                </Text>
+              </Box>
+            </Box>
+          )}
+        </DialogContent>
+        <DialogFooter>
+          <ButtonSecondary onClick={onClose}>Close</ButtonSecondary>
+        </DialogFooter>
+      </Flex>
+    </Dialog>
+  );
+}
+
+const requiredField = (value: string, fieldName: string) => () => {
+  if (!value || value.length === 0) {
+    return {
+      valid: false,
+      message: `${fieldName} is required`,
+    };
+  }
+
+  return {
+    valid: true,
+  };
+};
```

Comment:
we already have a rule for [required fields](https://github.com/gravitational/webapps/blob/master/packages/shared/components/Validation/rules.js#L18)?

Reply:
Ah, that's good to know, thanks!